pkg/registry/core/rest/storage_core.go

```diff
@@ -343,30 +347,38 @@ func (s componentStatusStorage) serversToValidate() map[string]*componentstatus.
 		"scheduler":          {Addr: "127.0.0.1", Port: ports.InsecureSchedulerPort, Path: "/healthz"},
 	}
 
+	if s.cloudProvider != nil && (len(s.cloudProvider.CloudProvider) != 0 || len(s.cloudProvider.CloudConfigFile) != 0) {
```

Comment:
The data flow of the component statuses API handler is backwards. The API server should not be making calls to its clients to report their health. We should definitely not add new instances of this anti-pattern, and should work to remove the existing instances of it. See https://github.com/kubernetes/enhancements/issues/553